backend/src/main/java/com/bakdata/conquery/resources/api/ConceptsProcessor.java

```diff
@@ -52,68 +52,82 @@
-			@Override
-			public FEList load(Concept<?> concept) throws Exception {
-				return FrontEndConceptBuilder.createTreeMap(concept);
-			}
-		});
-		
-	public FERoot getRoot(NamespaceStorage storage, User user) {
+																		   .softValues()
+																		   .expireAfterWrite(10, TimeUnit.MINUTES)
+																		   .build(new CacheLoader<Concept<?>, FEList>() {
+																			   @Override
+																			   public FEList load(Concept<?> concept) throws Exception {
+																				   return FrontEndConceptBuilder.createTreeMap(concept);
+																			   }
+																		   });
 
+	public FERoot getRoot(NamespaceStorage storage, User user) {
 		return FrontEndConceptBuilder.createRoot(storage, user);
 	}
-	
+
 	public FEList getNode(Concept<?> concept) {
 		try {
 			return nodeCache.get(concept);
 		}
 		catch (ExecutionException e) {
-			throw new RuntimeException("failed to create frontend node for "+concept, e);
+			throw new RuntimeException("failed to create frontend node for " + concept, e);
 		}
 	}
-	
+
 	public List<IdLabel> getDatasets(User user) {
 		return namespaces
-			.getAllDatasets()
-			.stream()
-			.filter(d -> user.isPermitted(DatasetPermission.onInstance(Ability.READ.asSet(), d.getId())))
-			.map(d -> new IdLabel(d.getLabel(), d.getId().toString()))
-			.sorted()
-			.collect(Collectors.toList());
+					   .getAllDatasets()
+					   .stream()
+					   .filter(d -> user.isPermitted(DatasetPermission.onInstance(Ability.READ.asSet(), d.getId())))
+					   .map(d -> new IdLabel(d.getLabel(), d.getId().toString()))
+					   .sorted()
+					   .collect(Collectors.toList());
 	}
 
 	/**
 	 * Search for all search terms at once, with stricter scoring.
 	 * The user will upload a file and expect only well-corresponding resolutions.
+	 *
+	 * @apiNote Users expect the results to be precise. Additionally they expect all potentially resolvable values to be returned. This is used to select a large list of entries in BigMultiSelects, where users cannot be expected to review every single value.
 	 */
 	public ResolvedConceptsResult resolveFilterValues(AbstractSelectFilter<?> filter, List<String> searchTerms) {
 
 		//search in the full text engine
-		Set<String> searchResult = createSourceSearchResult(filter.getSourceSearch(), searchTerms, filter.getSearchType()::score)
-										   .stream()
-										   .map(FEValue::getValue)
-										   .collect(Collectors.toSet());
+		Set<String> searchResult = Collections.emptySet();
+		QuickSearch<FilterSearchItem> search = filter.getSourceSearch();
+
+		// If we have a QuickSearch, use that for resolving first.
+		if (search != null) {
+			List<FilterSearchItem> result = search.findAllItems(searchTerms, filter.getSearchType()::score);
+
+			List<FEValue> res = result.stream()
+									  .map(item -> new FEValue(item.getLabel(), item.getValue(), item.getTemplateValues(), item.getOptionValue()))
+									  .collect(Collectors.toList());
```

Comment:
Es ist glaube ich nicht notwendig das suchergebis noch in ein  `FEValue` zu wrappen

Reply:
um es dann in der nächsten zeile wieder raus zuholen

Reply:
good catch!